src/qutipinterface.jl

```diff
@@ -0,0 +1,192 @@
+        evolved_state[i] = rho
+        i = i + 1
+    end
+    return time, evolved_state
+end
+
+mesolve(ME::QuLindbladMasterEq, rho0::QuBase.AbstractQuMatrix, tlist; solver=QuExpmV()) = mesolve(ME.hamiltonian, rho0, tlist, ME.collapse_ops; solver=solver)
+
+function mesolve(H::QuBase.AbstractQuArray, rho0::QuBase.AbstractQuMatrix, tlist, c_ops::Vector, eops::Vector; solver=QuExpmV())
+    expectation = zeros(Complex128, length(tlist)-1, length(eops))
+    j = 1
+    for (t,rho) in QuStateEvolution(H, c_ops, rho0, tlist, solver)
+        for i=1:length(eops)
+            expectation[j,i] = expectationvalue(rho, eops[i])
+        end
+        j = j + 1
+    end
+    return collect(tlist)[2:end], expectation
+end
+
+mesolve(ME::QuLindbladMasterEq, rho0::QuBase.AbstractQuMatrix, tlist, eops::Vector; solver=QuExpmV()) = mesolve(ME.hamiltonian, rho0, tlist, ME.collapse_ops, eops; solver=solver)
+
+
+@doc """
+Interface for `mcsolve`[http://qutip.org/docs/3.1.0/apidoc/functions.html#module-qutip.mcsolve] of QuTiP in QuDynamics
+
+`mcsolve` in QuDynamics, extended version.
+
+### Arguments
+
+Inputs :
+* H/ME :: AbstractQuArray/QuLindbladMasterEq Types
+
+  This parameter supports Hamiltonian of the system, Lindblad Master Equation
+* psi :: AbstractQuArray
+
+  Initial state vector of the system
+* tlist
+
+  Time step array/range
+* e_ops
+
+  Operators whose expectation values are to computed in the evolved states
+* solver
+
+  Keyword for selecting the solver from the various solvers supported by QuDynamics
+
+Output :
+* Array of tuples, which specify the time-step and the corresponding evolved state
+""" ->
+function mcsolve(H::QuBase.AbstractQuArray, psi0::QuBase.AbstractQuArray, tlist, c_ops::Vector; ntraj=1000)
+    time = collect(tlist)[2:end]
+    qumcwfen = QuMCWFEnsemble(complex(psi0), ntraj)
+    rho = complex(psi0*psi0')
+    rhos = Array(typeof(rho), length(tlist)-1)
+    for i=1:length(tlist)-1
+        rhos[i] = complex(zeros(rho))
+    end
+    for psi0 in qumcwfen
+        i = 1
+        for (t,psi) in QuStateEvolution(H, c_ops, psi0, tlist, QuMCWF())
+            rhos[i] = rhos[i] + (psi*psi')/length(qumcwfen)/norm(psi)^2
+            i = i + 1
+        end
+    end
+    return time, rhos
+end
+
+mcsolve(ME::QuLindbladMasterEq, psi0::QuBase.AbstractQuArray, tlist; ntraj=1000) = mcsolve(ME.hamiltonian, psi0, tlist, ME.collapse_ops; ntraj=ntraj)
+
+function mcsolve(H::QuBase.AbstractQuArray, psi0::QuBase.AbstractQuArray, tlist, c_ops::Vector, eops::Vector; ntraj=1000)
+    time = collect(tlist)[2:end]
+    qumcwfen = QuMCWFEnsemble(complex(psi0), ntraj)
+    expectation = zeros(Complex128, length(tlist)-1, length(eops))
+    for psi0 in qumcwfen
+        j = 1
+        for (t,psi) in QuStateEvolution(H, c_ops, psi0, tlist, QuMCWF())
+            for i=1:length(eops)
+                expectation[j,i] = expectation[j,i] + expectationvalue((psi*psi')/length(qumcwfen)/norm(psi)^2, eops[i])
```

Comment:
I would just use `psi` in `expectationvalue`and then divide it by `length(qumcwfen)*norm(psi)^2` afterwards. This way you only divide numbers instead of creating a matrix and then dividing all elements by the normalization.